ddclient

```diff
@@ -2075,6 +2176,37 @@ sub geturl {
 	}
 	$0 = sprintf("%s - closed %s port %s", $program, $peer, $port);
 
+    } else
+    {
+        # Access network using cURL.
+        my $curlopt = '';
+        my $protocol = 'http';
+        $curlopt = '--ipv4' if ($ipversion eq '4');
+        $curlopt = '--ipv6' if ($ipversion eq '6');
+        if ($use_ssl) {
+            $protocol = 'https';
+            $curlopt .= ' --insecure';
+        }
+
+        if (! opt('exec')) {
+	    debug("skipped network connection");
+	    verbose("SENDING:", "%s", '${protocol}://${server}/${url}');
+        } else
+        {
+            $0 = sprintf("%s - curl sending to %s", $program, '${protocol}://${server}/${url}');
+            my $timeout = opt('timeout');
+
+            $reply = `/usr/bin/curl -si0 --user "${login}:${password}" --user-agent "${program}/${version}" \\
+              --connect-timeout $timeout --max-time $timeout $curlopt \\
+              --url "${protocol}://${server}/${url}" 2>/dev/null`;
+
```

Comment:
Excellent idea to optionally use ```curl``` as it is ubiquitous, performs either IPv4/IPv6 and SSL all in one call.

But, may I suggest you use "HERE Documents" to minimize unwanted special-character processing:
````
	$reply = <<`USE_CURL`;
/usr/bin/curl -si0 --user '${login}:${password}' --user-agent '${program}/${version}' \\
	  --connect-timeout $timeout --max-time $timeout $curlopt \\
	  --url '${protocol}://${server}/${url}' 2>/dev/null
USE_CURL
```
Also use singe-quotes around the string arguments, better to allow ${password} special characters and does not effect the ${foo} perl replacement.